components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java

```diff
@@ -79,260 +56,95 @@
     @BeforeMethod
     public void setUp() throws Exception {
         oAuth2ScopeService = new OAuth2ScopeService();
-        CommonTestUtils.initPrivilegedCarbonContext();
+        IdentityUtil.populateProperties();
+    }
 
-        Field field = OAuth2ScopeService.class.getDeclaredField("scopeMgtDAO");
-        field.setAccessible(true);
-        field.set(oAuth2ScopeService, mockedScopeMgtDAO);
-        field.setAccessible(false);
+    @AfterMethod
+    public void tearDown() throws Exception {
+        Whitebox.setInternalState(IdentityUtil.class, "configuration", new HashMap<>());
+        Whitebox.setInternalState(IdentityUtil.class, "eventListenerConfiguration", new HashMap<>());
+        Whitebox.setInternalState(IdentityUtil.class, "identityCacheConfigurationHolder", new HashMap<>());
+        Whitebox.setInternalState(IdentityUtil.class, "identityCookiesConfigurationHolder", new HashMap<>());
     }
 
     @Test
     public void testRegisterScope() throws Exception {
-        String name = "dummyScopeName";
-        String description = "dummyScopeDescription";
-        mockedScope = new Scope(name, description);
-
-        mockStatic(IdentityUtil.class);
-        when(IdentityUtil.getIdentityCacheConfig(anyString(), anyString())).thenReturn(mockedIdentityCacheConfig);
-        when(mockedScopeMgtDAO.isScopeExists(anyString(), anyInt())).thenReturn(false);
 
-        Scope scope = oAuth2ScopeService.registerScope(mockedScope);
-        assertEquals(scope.getName(), "dummyScopeName", "Expected name did not received");
-        assertEquals(scope.getDescription(), "dummyScopeDescription", "Expected description did not received");
+        Scope dummyScope = new Scope(SCOPE_NAME, SCOPE_DESCRIPTION);
+        Scope scope = oAuth2ScopeService.registerScope(dummyScope);
+        assertEquals(scope.getName(), SCOPE_NAME, "Expected name did not received");
+        assertEquals(scope.getDescription(), SCOPE_DESCRIPTION, "Expected description did not received");
+        oAuth2ScopeService.deleteScope(SCOPE_NAME);
     }
 
     @Test(expectedExceptions = IdentityException.class)
     public void testRegisterScopeWithNoScopeName() throws Exception {
         String name = "";
         String description = "dummyScopeDescription";
-        mockedScope = new Scope(name, description);
-
-        oAuth2ScopeService.registerScope(mockedScope);
+        Scope scope = new Scope(name, description);
+        oAuth2ScopeService.registerScope(scope);
```

Comment:
Shall we assert for a value after calling get scope. Else it would only break if it throw an exception. Applicable for all test methods

Reply:
Hi Danushka, if this method doesn't throw an exception, it is a test failure. Suppose if we try to assert whether scope is there using getScope method,  and if it is there or not, still we are expecting to throw the exception. That means it is still a test failure.